modules/Core/hooks/MentionsHook.php

```diff
@@ -18,74 +18,76 @@ class MentionsHook extends HookBase {
      * - Replaces @mentions with [user] tags.
      * - Sends notifications to mentioned users.
      */
-    public static function preCreate(array $params = []): array {
-        if (self::validate($params)) {
-            $params['content'] = MentionsParser::parseAndNotify(
-                $params['user']->data()->id,
-                $params['content'],
-                $params['alert_url'],
-                $params['mention_notification_type'],
-                $params['mention_notification_title'],
-            );
+    public static function preCreate(AbstractEvent $event): void {
+        if (!empty($event->content) && isset($event->user)) {
+            if (isset($event->alert_url, $event->mention_notification_type, $event->mention_notification_title)) {
```

Comment:
Hmm, don't love all these required `isset` here... what do you think about some builtin event types such as `AbstractContentEvent extends AbstractEvent`?

Reply:
I don't get it you want a new event `AbstractContentEvent extends AbstractEvent`??

Or a class that the event extends/implements, However that class only only contain content being much having one for Content, User then alerts, and one for skip_content its all so different

Also if we do class then its bigger chance for future breaking changes for modules if we need to change the class